contracts/src/token/erc1155/mod.rs

```diff
@@ -0,0 +1,756 @@
+    }
+
+    fn safe_transfer_from(
+        &mut self,
+        from: Address,
+        to: Address,
+        token_id: U256,
+        value: U256,
+        data: Bytes,
+    ) -> Result<(), Self::Error> {
+        Ok(())
+    }
+
+    fn safe_batch_transfer_from(
+        &mut self,
+        from: Address,
+        to: Address,
+        token_ids: Vec<U256>,
+        values: Vec<U256>,
+        data: Bytes,
+    ) -> Result<(), Self::Error> {
+        Ok(())
+    }
+}
+
+impl Erc1155 {
+    /// Transfers a `value` amount of tokens of type `token_ids` from `from` to
+    /// `to`. Will mint (or burn) if `from` (or `to`) is the zero address.
+    ///
+    /// Requirements:
+    ///
+    /// * - If `to` refers to a smart contract, it must implement either
+    ///   [`IERC1155Receiver::on_erc_1155_received`]
+    /// * or [`IERC1155Receiver::on_erc_1155_received`] and return the
+    ///   acceptance magic value.
+    /// * - `token_ids` and `values` must have the same length.
+    ///
+    /// /// # Errors
+    ///
+    /// If length of `token_ids` is not equal to length of `values`, then the
+    /// error [`Error::InvalidArrayLength`] is returned.
+    /// If `value` is greater than the balance of the `from` account,
+    /// then the error [`Error::InsufficientBalance`] is returned.
+    ///
+    /// NOTE: The ERC-1155 acceptance check is not performed in this function.
+    /// See [`Self::_updateWithAcceptanceCheck`] instead.
+    ///
+    /// Event
+    ///
+    /// Emits a [`TransferSingle`] event if the arrays contain one element, and
+    /// [`TransferBatch`] otherwise.
+    fn _update(
+        &mut self,
+        from: Address,
+        to: Address,
+        token_ids: Vec<U256>,
+        values: Vec<U256>,
+    ) -> Result<(), Error> {
+        if token_ids.len() != values.len() {
+            return Err(Error::InvalidArrayLength(ERC1155InvalidArrayLength {
+                ids_length: Uint::<256, 4>::from(token_ids.len()),
+                values_length: Uint::<256, 4>::from(values.len()),
+            }));
+        }
+
+        let operator = msg::sender();
+        for (i, &token_id) in token_ids.iter().enumerate() {
+            let value = values[i];
+            let from_balance = self._balances.get(token_id).get(from);
+            if from_balance < value {
+                return Err(Error::InsufficientBalance(
+                    ERC1155InsufficientBalance {
+                        sender: from,
+                        balance: from_balance,
+                        needed: value,
+                        token_id,
+                    },
+                ));
+            }
+            self._balances
+                .setter(token_id)
+                .setter(from)
+                .sub_assign_unchecked(value);
+
+            if to != Address::ZERO {
+                self._balances
+                    .setter(token_id)
+                    .setter(to)
+                    .add_assign_unchecked(value);
```

Comment:
Calculate previously and panic when overflow like here https://github.com/OpenZeppelin/rust-contracts-stylus/blob/v0.1.0/contracts/src/token/erc20/mod.rs#L432-L435